include/seqan3/argument_parser/validators.hpp

```diff
@@ -485,6 +485,98 @@ struct default_validator
     }
 };
 
+//!\brief A helper struct to chain validators recursively via operator+()
+//!\ingroup validators
```

Comment:
A documentation about the fact, that both value types must be the same would be good.
And is it required that they are the same or can we use a less strict concept, such as implicit_convertible_to_concept?

Reply:
Or better they must fulfill `CommonReference<T, U>` ...

Reply:
My thoughts on this:

Before going into the problems, note that your proposed change would require another change for consistency: Currently in the `add_option(value, ..., validator)` call, the value type is required to be of the same type as the `validator::value_type` and and the functor of each validator has one parameter of that value_type. If we would relax this requirement to common reference type consider the following:

```
uint32_t val{};
arithmetic_range_validator<uint8_t> validator(3,10);
parser.add_option(val, ..., validator);
```
now the common type of uint8_t and uint32_t is uint32_t correct? so this code would be valid under your requirement. But comparing `val` to `uint8_t min = 3` will either trigger a compiler warning or probably yield wrong results, no? This could be solved by requiring that the type of `val` has to be implicelty castable to validator::value_type but I am not 100%sure if this would be a safe relaxation.

Now, if you chain validators a new validator with a value_type member is created. If I would only require that validator1 and validator2 hava a common reference, I would need to choose the common reference type for the new value_type, but I am not sure if this introduces complications when the adaptor is used in an `add_option` call.

Writing this down I think we could take the common reference type when chaining validators and then require the given value to be implicelty castable to the validator::value_type.... what do you think?

Reply:
@rrahn After discussing the issue can we leave this like it is for now? If so please re-review as I introduced all the other changes you requested besides this one and the common reference type below